test/methods.js

```diff
@@ -618,6 +618,55 @@ test( "remote, highlight all invalid fields", function( assert ) {
 		done();
 	}, 500 );
 } );
+test( "remote, unhighlighted should be invoked after being highlighted/invalid", function( assert ) {
+	expect( 2 );
+
+	var done1 = assert.async(),
+		done2 = assert.async(),
+		$form = $( "#testForm25" ),
+		$somethingField = $form.find( "input[name='something25']" ),
+		responseText = "false",
+		response = function() { return responseText; },
+		validateOptions = {
+			highlight: function( e ) {
+				$( e ).addClass( "error" );
+			},
+			unhighlight: function( e ) {
+				$( e ).removeClass( "error" );
+			},
```

Comment:
I stumbled upon the bug in our project when our unhighlight method override was not invoked after a successful remote validation if the field was in an invalid state. This test mimicks this behaviour.